src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/DataSourceParsingHandler.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient.SNI;
+using Microsoft.Data.SqlClientX.Handlers;
+using static Microsoft.Data.SqlClient.SNINativeMethodWrapper;
+
+namespace Microsoft.Data.SqlClientX.Handlers
+{
+    /// <summary>
+    /// Handler to populate data source information required for transport handler
+    /// </summary>
+    internal class DataSourceParsingHandler : IHandler<ConnectionHandlerContext>
+    {
+        /// <inheritdoc />
+        public IHandler<ConnectionHandlerContext> NextHandler { get; set; }
+
+        /// <inheritdoc />
+        public async ValueTask Handle(ConnectionHandlerContext request, bool isAsync, CancellationToken ct)
+        {
+            ServerInfo serverInfo = new ServerInfo(request.ConnectionString);
+            string fullServerName = serverInfo.ExtendedServerName;
+            string localDBDataSource = GetLocalDBDataSource(fullServerName, out bool errorWithLocalDBProcessing);
+
+            if (errorWithLocalDBProcessing)
+            {
+                //TODO: populate other details for the SQLException
+                SqlErrorCollection collection = new SqlErrorCollection();
+                collection.Add(new SqlError(0,0,TdsEnums.FATAL_ERROR_CLASS, null, StringsHelper.GetString("LocalDB_UnobtainableMessage"), null, 0));
```

Comment:
NIt: This could use a bit of formatting. e.g. space after `,` in the `new SqlError` parameters